tests/unit/test_distinguished_name.cpp

```diff
@@ -101,3 +101,32 @@ TEST_F(DistinguishedNameTest, testThatDistinguishedNameIsPopulatedCorrectly)
 
     ASSERT_THAT(dn, Eq(_dn));
 }
+
+TEST_F(DistinguishedNameTest, testThatCustomAttributeOrderWorks) {
+    auto customOrderBuilder = DistinguishedName::CustomOrderBuilder();
+    customOrderBuilder.commonName("ImATeapot")
+            .countryName("DE")
+            .pkcs9EmailAddress("support@example.com")
+            .organizationalUnitName("Linux Support")
+            .localityName("oben")
+            .serialNumber("08E36DD501941432358AFE8256BC6EFD")
+            .organizationName("Linux AG")
+            .stateOrProvinceName("nebenan");
+    auto dn = customOrderBuilder.build();
+    SSL_X509_NAME_Ptr x509Name = _X509_NAME_new();
+    dn.populateX509Name(x509Name);
+    ASSERT_EQ(_X509_NAME_get_index_by_NID(
+        x509Name.get(), openssl::ASN1_NID::CountryName)[0], 1);
+    ASSERT_EQ(_X509_NAME_get_index_by_NID(
+        x509Name.get(), openssl::ASN1_NID::Pkcs9EmailAddress)[0], 2);
+    ASSERT_EQ(_X509_NAME_get_index_by_NID(
+        x509Name.get(), openssl::ASN1_NID::OrganizationalUnitName)[0], 3);
+    ASSERT_EQ(_X509_NAME_get_index_by_NID(
+        x509Name.get(), openssl::ASN1_NID::LocalityName)[0], 4);
+    ASSERT_EQ(_X509_NAME_get_index_by_NID(
+        x509Name.get(), openssl::ASN1_NID::SerialNumber)[0], 5);
+    ASSERT_EQ(_X509_NAME_get_index_by_NID(
+        x509Name.get(), openssl::ASN1_NID::OrganizationName)[0], 6);
+    ASSERT_EQ(_X509_NAME_get_index_by_NID(
+        x509Name.get(), openssl::ASN1_NID::StateOrProvinceName)[0], 7);
+}
```

Comment:
There should be a test of the order on the old builder format is unchanged. I think there is no test right now that tests anything w.r.t. the order of elements.